implemented rest of OpenAPI Info Object

iancleary opened the pull request:
This pull request address issue https://github.com/kennethreitz/responder/issues/242

I have implemented the rest of the OpenAPI Info Object, added to the doc strings of the responder.api object, and updated the _apispec function.

I copied the names of the OpenAPI object exactly per https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#infoObject

I wasn't sure of your (the maintainers) preference on what to include in the docstring for the responder.api object. I figured it was fair to add information regarding the OpenAPI parameters, include the examples, and direct users that these variables map to the OpenAPI objects.  I went back and forth on if the examples were duplication, so if there are examples added to https://github.com/kennethreitz/responder/tree/master/examples, the docstring examples are probably redundant and misplaced.

Let me know if you like or dislike the typehinting in the doc strings. I am not sure if there is a convention there, but thought it was useful.

Thanks and please let me know if there is any updates you would like.

taoufik07 commented:
It fails because of the linting, use black to format

iancleary commented:
I've run black to format the file

taoufik07 commented:
Are you sure APISpec accept an info dict ?

taoufik07 commented:
Snake case for termsOfService

iancleary commented:
Hello,

APISpec accepts kwargs with the name options (**option). In the to_dict function this does a deep update. Since title and version are required, they are named parameters.  I used "info" as the name since the APISpec Info Object contains title and version, so that deepupdate in the to_dict function updates the correct dict and it propagates to the "to_yaml" function.

I've pushed commits for "Use is not None" and "Snake case for termsOf Service" (with the note that the dict in ApiSpec has to match the spec, but the responder.api variable will be snake case)

p.s. This pull request only addresses the Info Object.  A future request could implement customization of other parts.  For example, the "default" above all of the routes.  I've seen examples where there are groups of named routes. I think there is possible discussion needed on how that is implemented, as there is a one to many grouping of routes with that part of the APISpec (and I'd request this pull request just be the Info Object).

taoufik07 commented:
Can you please update the [docs](https://python-responder.org/en/latest/tour.html#openapi-schema-support) before merging ?

iancleary commented:
OK, will do.  I will update the  [docs](https://python-responder.org/en/latest/tour.html#openapi-schema-support).

Do you think the [test_documentation pytest function](https://github.com/kennethreitz/responder/blob/master/tests/test_responder.py#L367-L397) should have the additional parameters as well?

taoufik07 commented:
Yep

iancleary commented:
Done. 

I've updated the order to match the display and updated the openapi version to match the latest [OpenAPI Specification version](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md).

Someone with write access will have to merge.  Thank you for the feedback.

taoufik07 commented:
Thanks :D

iancleary commented:
You're welcome.  I'm happy to be a contributor.
